lib/dom_utils.coffee

```diff
@@ -315,6 +315,25 @@ DomUtils =
         false
     handlerStack.suppressEvent
 
+  consumeKeyup: (event, callback = null) ->
+    @suppressEvent event
+    keyChar = KeyboardUtils.getKeyCharString event
+    handlerStack.push
+      _name: "dom_utils/consumeKeyup"
+      keydown: (event) ->
+        @remove()
+        handlerStack.continueBubbling
+      keyup: (event) ->
+        return handlerStack.continueBubbling unless keyChar == KeyboardUtils.getKeyCharString event
+        @remove()
+        handlerStack.suppressEvent
+      # We cannot track keyup events if we lose the focus.
+      blur: (event) ->
+        @remove() if event.target == window
+        handlerStack.continueBubbling
+    callback?()
+    handlerStack.suppressEvent
+
```

Comment:
Will this not leak `keyup`s when multiple keys end up being held at the same time?

For example:
- typing `F`,
  * `[keydown shift][keydown f][keyup shift][keyup f]`
  * the `keyup` for `f`leaks
- typing `zH`,
  * `[keydown z][keydown shift][keyup z][keydown h][keyup shift][keyup h]`
  * slightly exaggerated example, but will leak the `keyup`s from `z` and `h` will both leak